src/ucp/tag/rndv.c

```diff
@@ -29,8 +29,9 @@ static int ucp_tag_rndv_is_get_op_possible(ucp_ep_h ep, uct_rkey_t rkey)
 
 static void ucp_rndv_rma_request_send_buffer_dereg(ucp_request_t *sreq)
 {
-    if ((UCP_DT_IS_CONTIG(sreq->send.datatype)) &&
-        (ucp_ep_is_rndv_lane_present(sreq->send.ep))) {
+    if (UCP_DT_IS_CONTIG(sreq->send.datatype) &&
+        ucp_ep_is_rndv_lane_present(sreq->send.ep) &&
+        ucp_request_is_send_buffer_reg(sreq) /* TODO: How is this possible? */) {
```

Comment:
so in this case, for consistency, need to change all places which set memh=NULL to set rsc=NULL, and replace the memh!=NULL by a check of rsc!=NULL